### PR TITLE
Refactor: event dispatcher

### PR DESCRIPTION
**Pull request by DimaDemchenko**

### CODE REVIEW

File: packages/p2p-media-loader-core/src/segments-storage.ts [hunk 1]
```diff
@@ -67,7 +69,7 @@ export class SegmentsMemoryStorage {
       lastAccessed: performance.now(),
     });
     this.logger(`add segment: ${id}`);
-    this.dispatchStorageUpdatedEvent(segment.stream);
+    this.onStorageUpdated(segment.stream);
```
Comment (mrlika) [inline packages/p2p-media-loader-core/src/segments-storage.ts:72]: This is different from the original code, which dispatches not `onStorageUpdated` here but `onStorageUpdated${StreamUtils.getStreamShortId(stream)}` event

**Review by mrlika — CHANGES_REQUESTED** [review]

File: packages/p2p-media-loader-core/src/segments-storage.ts [hunk 2]
```diff
@@ -150,19 +150,22 @@ export class SegmentsMemoryStorage {
...
+      `onStorageUpdated-${localId}`,
+      listener,
+    );
   }
 
   private dispatchStorageUpdatedEvent(stream: Stream) {
-    this.events.dispatch(
+    this.eventEmitter.dispatchEvent(
       `onStorageUpdated${StreamUtils.getStreamShortId(stream)}`,
       stream,
```
Comment (mrlika) [inline packages/p2p-media-loader-core/src/segments-storage.ts:170]: Test and fix

**Review by mrlika — CHANGES_REQUESTED** [review]

File: packages/p2p-media-loader-core/src/segments-storage.ts (hunk 2 above)
Comment (DimaDemchenko) [reply packages/p2p-media-loader-core/src/segments-storage.ts]: I fixed mistake and now it looks like it works properly
```
 this.segmentStorage.subscribeOnUpdate(
      this.stream,
      this.broadcastAnnouncement,
    );

    this.segmentStorage.subscribeOnUpdate(this.stream, this.testListener);

    this.trackerClient.start();
  }

  testListener = () => {
    console.log("test");
  };

```
image

File: packages/p2p-media-loader-core/src/segments-storage.ts [hunk 3]
```diff
@@ -150,20 +150,23 @@ export class SegmentsMemoryStorage {
     listener: StorageEventHandlers["onStorageUpdated"],
   ) {
     const localId = StreamUtils.getStreamShortId(stream);
-    this.events.subscribe(`onStorageUpdated-${localId}`, listener);
+    this.eventEmitter.addEventListener(`onStorageUpdated-${localId}`, listener);
   }
 
   unsubscribeFromUpdate(
     stream: Stream,
     listener: StorageEventHandlers["onStorageUpdated"],
```
Comment (mrlika) [inline packages/p2p-media-loader-core/src/segments-storage.ts:158]:
```suggestion
    listener: StorageEventHandlers["onStorageUpdated-"],
```

**Review by mrlika — CHANGES_REQUESTED** [review]

**Review by mrlika — APPROVED** [review]

**Review by mrlika — APPROVED** [review]

**Pull request merged**